### PR TITLE
add support for custom augmentations

### PR DESCRIPTION
**Pull request by vaskers5**
I add faced the same [issue](https://github.com/ultralytics/yolov5/issues/303) and support for custom torchvision.Transforms like augmentations

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced YOLOv5 image augmentation with torchvision transformations during inference.

### 📊 Key Changes
- Introduced `torchvision.transforms` as `T` to allow advanced image transformations.
- Modified the `forward` function signature to accept a transform composition (`augment: T.Compose`) instead of a boolean flag.
- Updated `_forward_augment` to apply the `augment` transformations to images before running inference.

### 🎯 Purpose & Impact
- **Purpose**: To enable more sophisticated and customizable image transformations during model inference, improving the versatility and accuracy of the model under various conditions.
- **Impact**: Users can now pass custom torchvision transform pipelines to the model during inference, potentially improving results and allowing for experiments with different augmentations. This change could lead to better performance in real-world applications by simulating various scenarios that the model may encounter.

### CODE REVIEW

**Comment by vaskers5** [comment]
now in can works like this:


`import torchvision.transforms as T

augment=T.Compose([
                   # T.ToTensor(),
                   # T.Normalize([0.5], [0.5]),
                   T.functional.rgb_to_grayscale,
                   lambda item: item.repeat(1, 3, 1, 1),
                   # T.ToPILImage()
                  ])

model(img_path,
      augment=augment
     ).show()`

**Comment by glenn-jocher** [comment]
@vaskers5 glad to know it's been working for you! YOLOv5 now supports custom augmentations with torchvision.transforms like this. If you have any other questions or need further assistance, feel free to ask!
